handle_actions.go

```diff
@@ -18,50 +19,57 @@ func (ms MocksArray) DoActions(c Context) error {
 	return nil
 }
 
+// DoActions checks if a given condition matches and performs each action in m.Actions
 func (m *Mock) DoActions(c Context) error {
 	c.Values = m.Values
 	if !c.MatchCondition(m.Expect.Condition) {
 		return nil
 	}
+
 	for _, actionDispatcher := range m.Actions {
 		actualAction := getActualAction(actionDispatcher)
-		if err := actualAction.Perform(c); err != nil {
-			logrus.WithFields(logrus.Fields{
-				"err":    err,
-				"action": fmt.Sprintf("%T", actualAction),
-			}).Errorf("failed to do action")
-		}
+		actualAction.Perform(c)
 	}
+
 	return nil
 }
 
-func (a ActionSendHTTP) Perform(context Context) error {
-	bodyStr, err := context.Render(a.Body)
-	if err != nil {
-		return err
-	}
+// Perform defines the action to perform for the ActionSendHTTP action
+func (a ActionSendHTTP) Perform(context Context) {
+	go func() {
+		if a.Sleep != 0 {
+			time.Sleep(a.Sleep)
+		}
 
-	urlStr, err := context.Render(a.URL)
-	if err != nil {
-		return err
-	}
+		bodyStr, err := context.Render(a.Body)
+		if err != nil {
+			handleActionErr(a, err)
+		}
 
-	request := gorequest.New().
-		SetDebug(true).
-		CustomMethod(a.Method, urlStr)
+		urlStr, err := context.Render(a.URL)
+		if err != nil {
+			handleActionErr(a, err)
+		}
 
-	for k, v := range a.Headers {
-		request.Set(k, v)
-	}
+		request := gorequest.New().
+			SetDebug(true).
+			CustomMethod(a.Method, urlStr)
 
-	_, _, errs := request.Send(bodyStr).End()
-	if len(errs) != 0 {
-		return errs[0]
-	}
-	return nil
+		for k, v := range a.Headers {
+			request.Set(k, v)
+		}
+
+		_, _, errs := request.Send(bodyStr).End()
+		if len(errs) != 0 {
+			if errs[0] != nil {
```

Comment:
Interesting that this can even happen.

Reply:
I had the same thought. assuming there is a case where the slice can be something like `[nil, err]`, but not 100% sure. don't want to break existing functionality